pymongo/pool.py

```diff
@@ -1483,33 +1483,33 @@ def connect(self, handler=None):
 
             raise
 
-        sock_info = SocketInfo(sock, self, self.address, conn_id)
+        connection = Connection(sock, self, self.address, conn_id)
         try:
             if self.handshake:
-                sock_info.hello()
-                self.is_writable = sock_info.is_writable
+                connection.hello()
+                self.is_writable = connection.is_writable
             if handler:
-                handler.contribute_socket(sock_info, completed_handshake=False)
+                handler.contribute_socket(connection, completed_handshake=False)
 
-            sock_info.authenticate()
+            connection.authenticate()
         except BaseException:
-            sock_info.close_socket(ConnectionClosedReason.ERROR)
+            connection.close_conn(ConnectionClosedReason.ERROR)
             raise
 
-        return sock_info
+        return connection
 
     @contextlib.contextmanager
-    def get_socket(self, handler=None):
-        """Get a socket from the pool. Use with a "with" statement.
+    def get_conn(self, handler=None):
```

Comment:
Since where changing this anyway, how do you feel about calling this "pool.checkout()" to closer to the CMAP spec?